### PR TITLE
SimpLL: Print info about analysis phases

### LINKED ISSUE #67: SimpLL: print info about analysis phases
For each phase of the analysis (pre-process, individual passes, analysis itself), write to stderr a message that the phase started. Indent all messages belonging to phases so that it is clear which message belongs to which phase.
For each phase and each pass, add a SimpLL CLI option to turn on/off messages for that phase. By default, only print messages for the analysis itself.
Remove the post-process phase which only does inlining that is not needed anymore.

### PR DESCRIPTION
**Pull request by Kuba314**
Resolves #67 
Adds these messages:
```
Analysing debugging information...
Preprocessing {mod_name}...
Slicing control-flow-only statements...
Generating function abstractions in module {mod_name}...
Removing unused return values in module {mod_name}...
Slicing {fun_name} w.r.t. value of {var_name}...
```

### CODE REVIEW

File: diffkemp/simpll/DebugInfo.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@ using namespace llvm;
 
 PreservedAnalyses RemoveDebugInfoPass::run(Function &Fun,
                                            FunctionAnalysisManager & /*fam*/) {
+    DEBUG_WITH_TYPE(DEBUG_SIMPLL,
+                    dbgs() << "Analysing debugging information...\n");
```
Comment (viktormalik) [inline diffkemp/simpll/DebugInfo.cpp:24]: This should go to `DebugInfo` (probably the best place would be its constructor).

File: diffkemp/simpll/passes/FunctionAbstractionsGenerator.cpp [hunk 2]
```diff
@@ -26,6 +26,9 @@ AnalysisKey FunctionAbstractionsGenerator::Key;
 /// and for each pair of assembly code and constraint.
 FunctionAbstractionsGenerator::Result FunctionAbstractionsGenerator::run(
         Module &Mod, AnalysisManager<Module, Function *> &mam, Function *Main) {
+    DEBUG_WITH_TYPE(DEBUG_SIMPLL,
+                    dbgs() << "Generating function abstractions in module "
```
Comment (viktormalik) [inline diffkemp/simpll/passes/FunctionAbstractionsGenerator.cpp:30]:
```suggestion
                    dbgs() << "Generating function abstractions in "
```
... same for other messages (the word "module" is redundant here).

**Review by viktormalik — CHANGES_REQUESTED** [review]
Nice, thanks!

I'd suggest one more improvement - for each pass where you added this message, indent all other debug messages printed by that pass by one level (you can use functions `*DebugIndent*` from `Utils.h`). Similarly, for passes that are run from `preprocessModule`, indent their main info message by one level (and, consequently, all their inner messages by 2 levels).

File: diffkemp/simpll/DebugInfo.cpp (hunk 1 above)
Comment (Kuba314) [reply diffkemp/simpll/DebugInfo.cpp:24]: Does that mean creating a destructor and putting `decreaseDebugIndentLevel()` in there?

Comment (viktormalik) [reply diffkemp/simpll/DebugInfo.cpp]: Most of the analysis is done in the constructor, so I'd put `decreaseDebugIndentLevel()` to its end.

**Comment by Kuba314** [comment]
@viktormalik Could you rerun CI? Don't know why it failed exactly, but it seems like it wasn't my fault.

**Review by viktormalik — APPROVED** [review]

**Comment by viktormalik** [comment]
Please rebase once the CI passes.

**Review by viktormalik — APPROVED** [review]

**Pull request merged**